shared/react-native/ios/Keybase/NativeLogger.m

```diff
@@ -27,20 +27,22 @@ + (BOOL)requiresMainQueueSetup
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)
 {
-  NSMutableArray<NSString *> *lines = [[NSMutableArray alloc] init];
-  AppDelegate *appDel = (AppDelegate *)[[UIApplication sharedApplication] delegate];
-  DDFileLogger *fileLogger = (DDFileLogger *)appDel.fileLogger;
-  NSArray<NSString *> *paths = [[fileLogger logFileManager] sortedLogFilePaths];
-  for (NSString *path in paths) {
-    NSString *fileContents = [NSString stringWithContentsOfFile:path encoding:NSUTF8StringEncoding error:NULL];
-    for (NSString *line in [fileContents componentsSeparatedByCharactersInSet:[NSCharacterSet newlineCharacterSet]]) {
-      NSRange range = [line rangeOfString:[NSString stringWithFormat:@"%@%@: ", tagPrefix, tagName]];
-      if (range.location != NSNotFound) {
-        [lines addObject:[line substringFromIndex:range.location + range.length]];
+  dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
this likely needs to only happen in the main queue. access to the AppDelegate for sure but likely this is all bad to touch from multiple threads

Reply:
Maybe better for this to have its own queue?

Reply:
@mmaxim done